sklearn/tree/_splitter.pyx

```diff
@@ -348,32 +388,84 @@ cdef class BestSplitter(BaseDenseSplitter):
             features[f_i], features[f_j] = features[f_j], features[f_i]
 
             # Evaluate all splits
-            self.criterion.reset()
-            p = start
+            self.criterion.init_missing(n_missing)
 
-            while p < end:
-                while p + 1 < end and Xf[p + 1] <= Xf[p] + FEATURE_THRESHOLD:
-                    p += 1
+            directions = 1 if n_missing == 0 else 2
+            for i in range(directions):
+                missing_go_to_left = i != 0
+                self.criterion.missing_go_to_left = missing_go_to_left
+                self.criterion.reset()
```

Comment:
Is this the part where if there are no missing samples, then they always go to the left?